### ISSUE TITLE
`cargo test` compilation error

### ISSUE THREAD

**Issue by rkudryashov** [testcontainers/testcontainers-rs#231]
In my multi-module project while I am running `cargo test` in the root directory, I'm getting:

```
>cargo test
   Compiling winapi v0.3.9
   Compiling wepoll-sys v3.0.0
   Compiling libz-sys v1.1.2
   Compiling cmake v0.1.44
   Compiling rdkafka-sys v2.1.0+1.5.0
   Compiling testcontainers v0.11.0
error[E0433]: failed to resolve: could not find `fmt` in `core`
  --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\testcontainers-0.11.0\src\core\docker.rs:74:10
   |
74 | #[derive(derivative::Derivative)]
   |          ^^^^^^^^^^^^^^^^^^^^^^ could not find `fmt` in `core`
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0433`.
error: could not compile `testcontainers`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

OS: Windows 10

There is no error while running tests in submodules. Also, there is no error if I use `0.9.1`.

**Comment by thomaseizinger** [comment]
Hi @rkudryashov !

Thanks for the report. Which version of `rustc` are you using? Is any other crate in your dependency graph using `derivative`?

The error seems to originate from the `derivative` crate which - now that I think about it - we probably shouldn't depend on and instead just create those impls ourselves :)

**Comment by thomaseizinger** [comment]
Can you check whether or not this PR fixes things for you? https://github.com/testcontainers/testcontainers-rs/pull/232

**Comment by rkudryashov** [comment]
> Which version of `rustc` are you using? Is any other crate in your dependency graph using `derivative`?

`rustc 1.47.0 (18bf6b4f0 2020-10-07)`

It seems like `rdkafka` depends on `derivative`:

```
├── rdkafka v0.24.0
│   ├── futures v0.3.6 (*)
│   ├── libc v0.2.79
│   ├── log v0.4.11 (*)
│   ├── rdkafka-sys v2.1.0+1.5.0
│   │   ├── libc v0.2.79
│   │   ├── libz-sys v1.1.2
│   │   │   └── libc v0.2.79
│   │   │   [build-dependencies]
│   │   │   ├── cc v1.0.60 (*)
│   │   │   ├── pkg-config v0.3.19
│   │   │   └── vcpkg v0.2.10
│   │   └── num_enum v0.5.1
│   │       ├── derivative v2.1.1
│   │       │   ├── proc-macro2 v1.0.24 (*)
│   │       │   ├── quote v1.0.7 (*)
│   │       │   └── syn v1.0.44 (*)
│   │       └── num_enum_derive v0.5.1
│   │           ├── proc-macro-crate v0.1.5 (*)
│   │           ├── proc-macro2 v1.0.24 (*)
│   │           ├── quote v1.0.7 (*)
│   │           └── syn v1.0.44 (*)
│   │   [build-dependencies]
│   │   ├── cmake v0.1.44
│   │   │   └── cc v1.0.60 (*)
│   │   └── pkg-config v0.3.19
│   ├── serde v1.0.117 (*)
│   ├── serde_derive v1.0.117 (*)
│   ├── serde_json v1.0.59 (*)
│   └── tokio v0.2.22 (*)
```

> Can you check whether or not this PR fixes things for you? #232

Thanks, tests are compiled using this branch

**Comment by thomaseizinger** [comment]
> Thanks, tests are compiled using this branch

Ok that is good to know! Unfortunately Travis is currently broken for an unknown reason and I haven't gotten around to fixing it yet so I can't merge that PR. Sorry for the delay!

**Comment by ctron** [comment]
I am running into the same issue. From what I saw this fix is not released yet. Is there a chance that this can be released soon?

**Comment by thomaseizinger** [comment]
Just published 0.12.0.
